llvm/docs/InstrProfileFormat.rst

```diff
@@ -0,0 +1,457 @@
+  between the metadata and bitmap section in the instrumented binary,
+  i.e., `start(__llvm_prf_bits) - start(__llvm_prf_data)`.
+
+  It's used jointly with the in-memory address difference of a profile data record
+  and its bitmap in the instrumented binary to find the bitmap of a profile data
+  record, in a similar way to how counters are referenced as explained by
+  calculation-of-counter-offset_ .
+
+``NamesDelta``
+  Records the in-memory address of name section. Not used except for raw profile
+  reader error checking.
+
+``ValueKindLast``
+  Records the number of value kinds. Macro `VALUE_PROF_KIND`_ defines the value
+  kinds with a description of the kind.
+
+.. _`VALUE_PROF_KIND`: https://github.com/llvm/llvm-project/blob/7e405eb722e40c79b7726201d0f76b5dab34ba0f/compiler-rt/include/profile/InstrProfData.inc#L184-L186
+
+Payload Sections
+------------------
+
+Binary Ids
+^^^^^^^^^^^
+Stores the binary ids of the instrumented binaries to associate binaries with
+profiles for source code coverage. See `Binary Id RFC`_ for the design.
+
+.. _`Binary Id RFC`: https://lists.llvm.org/pipermail/llvm-dev/2021-June/151154.html
+
+Profile Metadata
+^^^^^^^^^^^^^^^^^^
+
+This section stores the metadata to map counters and value profiles back to
+instrumented code regions (e.g., LLVM IR for IRPGO).
+
+The in-memory representation of the metadata is `__llvm_profile_data`_.
+Some fields are used to reference data from other sections in the profile.
+The fields are documented as follows:
+
+.. _`__llvm_profile_data`: https://github.com/llvm/llvm-project/blob/7c3b67d2038cfb48a80299089f6a1308eee1df7f/compiler-rt/lib/profile/InstrProfiling.h#L25
+
+``NameRef``
+  The MD5 of the function's PGO name. PGO name has the format
+  `[<filepath><delimiter>]<linkage-or-mangled-name>` where `<filepath>` and
+  `<delimiter>` is provided for local-linkage functions to tell possibly
+  identical functions.
+
+.. _FuncHash:
+
+``FuncHash``
+  A checksum of the function's IR, taking control flow graph and instrumented
+  value sites into accounts. See `computeCFGHash`_ for details.
+
+.. _`computeCFGHash`: https://github.com/llvm/llvm-project/blob/7c3b67d2038cfb48a80299089f6a1308eee1df7f/llvm/lib/Transforms/Instrumentation/PGOInstrumentation.cpp#L616-L685
+
+``CounterPtr``
+  The in-memory address difference between profile data and the start of corresponding
+  counters. Counter position is stored this way (as a link-time constant) to reduce
+  instrumented binary size compared with snapshotting the address of symbols directly.
+  See `commit a1532ed`_ for further information.
+
+.. _`commit a1532ed`: https://github.com/llvm/llvm-project/commit/a1532ed27582038e2d9588108ba0fe8237f01844
+
+  .. note::
+    `CounterPtr` might represent a different value for non-IRPGO use case. For
+    example, for `binary profile correlation`_, it represents the counter address.
+    When in doubt, check source code.
+
+``BitmapPtr``
```

Comment:
Similar to CounterPtr, BitmapPtr is also an absolute address of the symbol under binary correlation.

Reply:
I update the description of the `CounterPtr` and `BitmapPtr`. PTAL.